library/src/main/scala-2.13/scala/scalajs/js/ImmutableArray.scala

```diff
@@ -0,0 +1,47 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-lang.org/     **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package scala.scalajs.js
+
+import scala.annotation.unchecked.uncheckedVariance
+import scala.collection.{IterableOnce, SeqFactory, StrictOptimizedSeqFactory, immutable, mutable}
+
+
+/** Equivalent of sci.ImmutableArray for js.Array */
+final class ImmutableArray[+A] private (array: Array[A])
```

Comment:
Note that there are no tests for this collection (apart from varargs usage in tests).